docs/session_state_api.md

```diff
@@ -0,0 +1,181 @@
+# Session State API
```

Comment:
Given our current setup with API docs on one page, why are the State API on a different page altogether?

Reply:
This is likely a Product decision, but I'm unsure if we're wedded to it (cc @asaini). There are 3 primary types of state that are important in Streamlit app development: Session State, Global State, and User State. 

Global and User State will be launched as separate features in the future. The idea would be to populate this State API page as and when those features are released.

Further, the API reference documents the API for streamlit elements. Like components, session state doesn't fall into this elements category and warrants its own page.